recorder/src/main/cpp/stacktraces.cc

```diff
@@ -0,0 +1,23 @@
+#include "stacktraces.hh"
+
+std::uint32_t Stacktraces::fill_backtrace(NativeFrame* buff, std::uint32_t capacity) {//TODO: write me 3 tests { (capacity > stack), (stack > capacity) and (stack == capacity) }
+    std::uint64_t rbp, rpc;
+    asm("movq %%rbp, %%rax;"
+        "movq %%rax, %0;"
+        "lea (%%rip), %%rax;"
```

Comment:
It'll mangle 2 registers, hence making compiler's life slightly more difficult, but it'll work. Yes, another overflow to L1 will happen and unfortunately if a ctx-sw happens between that overflow and asm-fragment end, the data may come from L3 or worse, DRAM, which will be very slow, but that is the case for so many computations anyway, this is no different.